api/v2/api.go

```diff
@@ -53,6 +53,8 @@ type API struct {
 	peer           *cluster.Peer
 	silences       *silence.Silences
 	alerts         provider.Alerts
+	muter          types.Muter
```

Comment:
If `muter` is only about inhibition, let's also call it that way. In addition, this could also just be a function. What do you think?
```suggestion
	isInhibited          func(model.LabelSet) bool
```

Reply:
There is already `type MuteFunc`. A function `f` of the suggested signature can be converted into a `Muter` by `MuteFunc(f)`. I'd say accepting the interface here (which could be satisfied by the function you suggest) is preferred.

But I'm with you WRT naming the member variable with something that suggests this is only about inhibitions.

Reply:
The `Muter` interface's description of its method, and its single implementater `Inhibitor`, is a bit confusing. Our implementation relies on a side effect of the particular impl from `Inhibitor`, and isn't actually checking if the alert is inhibited -- instead, if it is, internal to the method, the alert's status is updated.

This is a long way of saying I'm thinking of naming the struct member something along the lines of `setInhibitedStatus`, and separately (an issue or something) discuss how we should either drop or update this interface.

Because we're relying on a particular implementation detail, I think the "correct" thing to do is NOT use the interface, but the actual concrete implementation.

Comments?

Reply:
Quite a bit has changed now. It looks good to me as it is now, but I leave the final call to @mxinden as he raised this point in the first place.